File: PDOO-Civitas/CivitasR/lib/test_p4.rb
#encoding: utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "jugador"
require_relative "jugador_especulador"
require_relative "titulo_propiedad"
require_relative "sorpresa_especulador"
require_relative "sorpresa"
require_relative "diario"
require_relative "mazo_sorpresas"

module Civitas
  class TestP4
    
    def self.main()
      
      jugador = Jugador.new("Ramon", nil)
        
      propiedad = Titulo_Propiedad.new("Ronda de Valencia", 10, 0.5, 25, 50, 20)
        
      propiedad.actualizar_propietario_por_conversion(jugador)
      jugador.propiedades.push(propiedad)
      
      jugadores = Array.new
      
      jugadores.push(jugador)

      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      puts jugadores.at(0).to_s

      puts "\n================================= HACEMOS LA CONVERSIÓN ==================================\n"

      sorpresa = Sorpresa_especulador.new(100, "Te conviertes en un especulador con 100€ de fianza")

      sorpresa.aplicar_a_jugador(0, jugadores)
        
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
        
      jugadores.at(0).construir_hotel(0)
      
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      jugadores.at(0).construir_casa(0)
      
      puts jugadores.at(0).to_s
    end
  end
  
  TestP4.main()
  
end

File: PDOO-Civitas/CivitasJ/src/civitas/TestP4.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ramonjtorres
 */
public class TestP4 {
    
    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
    
        Jugador jugador = new Jugador("Ramon");
        
        TituloPropiedad propiedad = new TituloPropiedad("Ronda de Valencia", 10, (float) 0.5,25,50,20);
        
        propiedad.actualizaPropietarioPorConversion(jugador);
        jugador.propiedades.add(propiedad);
        
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        
        jugador.construirHotel(0);
        
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        
        jugador.construirHotel(0);
        
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        
        jugador.construirHotel(0);
        
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        
        jugador.construirHotel(0);
        
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        
        jugador.construirHotel(0);
        
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        jugador.construirCasa(0);
        
        jugador.construirHotel(0);
        
        ArrayList<Jugador> jugadores = new ArrayList<Jugador>();
        
        jugadores.add(jugador);
        
        System.out.println(jugador.toString());
        
        System.out.println("\n================================= HACEMOS LA CONVERSIÓN ==================================\n");
        
        SorpresaEspeculador sorpresa = new SorpresaEspeculador(100, "Te conviertes en un especulador con 100€ de fianza");
        
        sorpresa.aplicarAJugador(0, jugadores);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        jugadores.get(0).construirCasa(0);
        
        jugadores.get(0).construirHotel(0);
        
        System.out.println(jugadores.get(0).toString());
        
    }
    
}
